src/actions/sources/select.js

```diff
@@ -96,87 +100,15 @@ export function selectSource(sourceId: string) {
  * @static
  */
 export function selectLocation(location: Location) {
-  return async ({ dispatch, getState, client }: ThunkArgs) => {
-    if (!client) {
-      // No connection, do nothing. This happens when the debugger is
-      // shut down too fast and it tries to display a default source.
-      return;
-    }
-
-    const source = getSource(getState(), location.sourceId);
-    if (!source) {
-      // If there is no source we deselect the current selected source
-      return dispatch(clearSelectedLocation());
-    }
-
-    const activeSearch = getActiveSearch(getState());
-    if (activeSearch !== "file") {
-      dispatch(closeActiveSearch());
-    }
-
-    dispatch(addTab(source.url, 0));
-    dispatch(setSelectedLocation(source, location));
-
-    await dispatch(loadSourceText(source));
-    const loadedSource = getSource(getState(), source.id);
-
-    if (!loadedSource) {
-      // If there was a navigation while we were loading the loadedSource
-      return;
-    }
-
-    if (
-      prefs.autoPrettyPrint &&
-      !getPrettySource(getState(), loadedSource.id) &&
-      shouldPrettyPrint(loadedSource) &&
-      isMinified(loadedSource)
-    ) {
-      await dispatch(togglePrettyPrint(loadedSource.id));
-      dispatch(closeTab(loadedSource.url));
-    }
-
-    dispatch(setSymbols(loadedSource.id));
-    dispatch(setOutOfScopeLocations());
-  };
+  return locationSelect(location, {checkPrettyPrint: true});
```

Comment:
looks good, maybe instead of `locatonSelect`, we add a param  here `selectLocation(location: Location, {checkPrettyPrint: true})`

also lets do this in a separate PR